src/Core/Helpers/InterfaceToConcreteTypeConverter.cs

```diff
@@ -0,0 +1,41 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using Newtonsoft.Json;
+
+namespace SonarLint.VisualStudio.Core.Helpers
+{
+    /// <summary>
+    /// Custom deserializer to help serializing/deserializing to types that have interface property types
+    /// </summary>
+    /// <typeparam name="TInterface">The interface type being serialized/deserialized</typeparam>
+    /// <typeparam name="TConcrete">The concrete type that NewtonSoft.Json should use</typeparam>
+    public class InterfaceToConcreteTypeConverter<TInterface, TConcrete> : JsonConverter where TConcrete : TInterface
+    {
+        public override bool CanConvert(Type objectType) => objectType == typeof(TInterface);
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+            => serializer.Deserialize<TConcrete>(reader);
+
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+            => serializer.Serialize(writer, value);
```

Comment:
I'll add a few simple tests for this; currently it's partially tested indirectly by the new `RuleInfoJsonDeserializer`.